cmd/commands/flatten_rbd_pvc.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RBDInfoOutput struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type Watcher struct {
+	Address string `json:"address"`
+}
+
+type RBDStatusOutput struct {
+	Watchers []Watcher `json:"watchers"`
+}
+
+var namespace string
+var allowInUse bool
+
+// FlattenRBDPVCCmd represents the rook commands
+var FlattenRBDPVCCmd = &cobra.Command{
+	Use:   "flatten-rbd-pvc",
+	Short: "Flatten the RBD image corresponding to the target RBD PVC",
+	Long: `Flatten the RBD image corresponding to the target RBD PVC.
+The target RBD PVC must be a cloned image and must be created by ceph-csi.
+This command removes the corresponding temporary cloned image[1]
+if the target PVC was cloned from another PVC.
+
+[1]: https://github.com/ceph/ceph-csi/blob/devel/docs/design/proposals/rbd-snap-clone.md`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
@satoru-takeuchi overall pr looks good, small suggestion if you could move the implementation function to under `pkg/` folder and not put it in the same file as where command is written. Thanks after that PR will be ready to merge.

Reply:
@subhamkrai Indeed, moved.

Reply:
merged